dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AliasableResourceInterface.java

```diff
@@ -56,32 +60,84 @@
+     * @param blockAliasesWithZenodoFormat block creation of an alias with a particular format
+     * @return the alias as a string
+     */
+    static void checkAliases(Set<String>  aliases, User user, boolean blockAliasesWithZenodoFormat) {
+        // Admins and curators do not have restrictions on alias format
+        if (user.isCurator() || user.getIsAdmin()) {
+            return;
+        }
+
+        checkAliasFormat(aliases, blockAliasesWithZenodoFormat);
+    }
 
+
+    /**
+     * Check that aliases do not contain invalid prefixes
+     * if the user adding them is not an admin or curator
+     * @param aliases a Set of alias strings
+     * @param blockAliasesWithZenodoFormat block creation of an alias with a particular format
+     * @return the alias as a string
+     */
+    static void checkAliasFormat(Set<String>  aliases, boolean blockAliasesWithZenodoFormat) {
         // Gather up any aliases that contain invalid prefixes
         List<String> invalidAliases = new ArrayList<>();
-        if (!user.isCurator() && !user.getIsAdmin()) {
-            invalidAliases = aliases.stream().filter(alias -> StringUtils.startsWithAny(alias, invalidPrefixes)).collect(Collectors.toList());
-        }
+        invalidAliases = aliases.stream().filter(alias -> StringUtils.startsWithAny(alias, INVALID_PREFIXES)).collect(Collectors.toList());
 
         // If there are any aliases with invalid prefixes then report it to the user
         if (invalidAliases.size() > 0) {
             String invalidAliasesString = String.join(", ", invalidAliases);
-            String invalidPrefixesString = String.join(", ", invalidPrefixes);
-            throw new CustomWebApplicationException("These aliases: " + invalidAliasesString + " start with a reserved string,"
+            String invalidPrefixesString = String.join(", ", INVALID_PREFIXES);
+            throw new CustomWebApplicationException("These aliases: " + invalidAliasesString + " start with a reserved string."
                     + " They cannot be used. Please create aliases without these prefixes: " + invalidPrefixesString,
                     HttpStatus.SC_BAD_REQUEST);
         }
+
+        if (blockAliasesWithZenodoFormat) {
+            Pattern pattern = Pattern.compile(ZENDO_DOI_REGEX);
+            List<String> aliasesWithForbiddenFormat = aliases.stream().filter(alias -> pattern.matcher(alias).matches()).collect(Collectors.toList());
+            // If there are any aliases with invalid formats then report it to the user
+            if (aliasesWithForbiddenFormat.size() > 0) {
+                String invalidAliasesString = String.join(", ", aliasesWithForbiddenFormat);
+                throw new CustomWebApplicationException("These aliases : " + invalidAliasesString + " have a format that is forbidden."
+                        + " They cannot be used. Please create aliases without this format: " + ZENDO_DOI_REGEX,
```

Comment:
I don't think a regex by itself like that is a very user-friendly message. Most people won't be able to make sense of it. Maybe add an example as well as the regex?

Reply:
Doesn't seem like it is really needed because Dockstore API displays the user's alias with the forbidden format, so the user already has an example of a forbidden format.

Reply:
Still seems odd to me to return the literal Java regex string, where you have to double escape things, e.g., `\\d` instead of `\d`.  But in practice only a malicious user is going to run into this, so it's not a big deal.